csi/volumeutils.py

```diff
@@ -914,7 +974,8 @@ def mount_glusterfs(volume, mountpoint, is_client=False):
             cmd.extend(["--client-pid", "-14"])
 
         try:
-            execute(*cmd)
+            (_, err, pid) = execute(*cmd)
+            VOL_DATA[volname]["pid"] = pid
```

Comment:
Question @aravindavk : looks like the problem here is, pid is of the parent process (ie, glusterfs binary), but the process does fork() so this PID we received may not be running when we want to issue the SIGHUP.

In that case, I would do the kill sighup with a piped command to get the volume process based on name etc.. ?

Reply:
thats good approach I think. Since the Storage names are unique, we can search the `ps` output to get the pid.

```
ps --no-header -ww -o pid,command -C glusterfs | grep "\-\-volfile-id=${volname}"
```

Reply:
```python
def get_volumes_mount_pids():
    cmd = ['ps', '--no-header', '-ww', '-o', 'pid,command', '-C', 'glusterfs']
    out, err = execute(*cmd)
    volumes = {}

    for line in out.split("\n"):
        parts = line.split()
        pid = parts[0]
        volname = ""
        for part in parts:
            if part.startswith("--volfile-id="):
                volname = part.split("=")[-1]

        volumes[volname] = pid

    return volumes
```